test/sql/test_name_check/R/test_name_check

```diff
@@ -0,0 +1,35 @@
+-- name: test_name_check
+create database `abc-~!@#$%^&<>=+abc`;
```

Comment:
I think should test some specail char start case

Reply:
btw, sr's interface `jdbc/mysqlclient/http` can use? like "@abc", "$abc", “&abc”, looks can't work well in http protocal(steram load) ?

Reply:
For now, we still need the start character is an alphabet or `_`

Reply:
`http://<fe_host>:<fe_http_port>/api/<database_name>/<table_name>/_stream_load`. Seems we need add some tests to verify the special character in <database_name>/<table_name>

Reply:
After testing,  if a URL contains special characters, it needs to be encoded using percent encoding(like ! -> %21). When parsing requests on the FE, the URL is decoded. However, when forwarding the request to the BE, it is re-encoded. The issue arises because the BE does not decode the request, leading to parsing errors for certain special characters, thus preventing normal stream loading. Therefore, the BE needs to add a decoding step similar to the front-end. I will record this issue to DW.